client/src/main/java/io/pravega/client/netty/impl/RawClient.java

```diff
@@ -145,6 +163,7 @@ private void closeConnection(Throwable exceptionToInflightRequests) {
                     reply.completeExceptionally(cfe);
                     closeConnection(cfe);
                 }
+                timer.completeExceptionally(new InterruptedException());
```

Comment:
I would think `cancel(true)` would be more appropriate?

Reply:
You are right, thanks.